src/main/java/org/elasticsearch/index/query/TermQuery.java

```diff
@@ -0,0 +1,205 @@
+     *  - code wise two classes are merged into one which is nice
+     *  - filter and query can maybe share once class and we add a `toFilter(QueryParserContenxt ctx)` method that returns a filter and by default return a `new QueryWrapperFilter(toQuery(context));`
+     */
+
+    public static final String NAME = "term";
+
+    private String fieldName;
+
+    private Object value;
+
+    private float boost = 1.0f;
+
+    private String queryName;
+
+    public TermQuery() {
+    }
+
+    public TermQuery(String fieldName, Object value) {
+        this.fieldName = fieldName;
+        this.value = value;
+    }
+
+    public TermQuery(String fieldName, Object value, String queryName) {
+        this.fieldName = fieldName;
+        this.value = value;
+        this.queryName = queryName;
+    }
+
+    public String getFieldName() {
+        return fieldName;
+    }
+
+    public Object getValue() {
+        return value;
+    }
+
+    public float getBoost() {
+        return boost;
+    }
+
+    public String getQueryName() {
+        return queryName;
+    }
+
+    public void setBoost(float boost) {
+        this.boost = boost;
+    }
+
+    public void setQueryName(String queryName) {
+        this.queryName = queryName;
+    }
+
+    @Override
+    public void readFrom(StreamInput in) throws IOException {
+        fieldName = in.readString();
+        value = in.readGenericValue();
+        boost = in.readFloat();
+        queryName = in.readOptionalString();
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeString(fieldName);
+        out.writeGenericValue(value);
+        out.writeFloat(boost);
+        out.writeOptionalString(queryName);
+    }
+
+    @Override
+    public void doXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject(NAME);
+        if (boost == 1.0f && queryName == null) {
+            builder.field(fieldName, value);
+        } else {
+            builder.startObject(fieldName);
+            builder.field("value", value);
+            if (boost != 1.0f) {
+                builder.field("boost", boost);
+            }
+            if (queryName != null) {
+                builder.field("_name", queryName);
+            }
+            builder.endObject();
+        }
+        builder.endObject();
+    }
+
+    @Override
+    public String[] names() {
+        return new String[]{TermQuery.NAME};
+    }
+
+    @Override
+    public Query parse(QueryParseContext parseContext) throws IOException, QueryParsingException {
+        TermQuery query = new TermQuery();
+        query.fromXContent(parseContext);
+        return query.toQuery(parseContext);
+    }
+
+    public void fromXContent(QueryParseContext context) throws IOException {
```

Comment:
I would feel much better making `fromXContent` and `toQuery` private here, otherwise I feel like it is a very "stateful" looking API, because if someone tries to use `toQuery` without calling `fromXContent` first they'll get exceptions.

Is there a reason they should be public?